File: utils/src/main/java/io/github/aemogie/timble/utils/logging/LoggerOutputStream.java
package io.github.aemogie.timble.utils.logging;

import org.jetbrains.annotations.NotNull;

import java.io.OutputStream;
import java.util.function.Consumer;

public class LoggerOutputStream extends OutputStream {
	private final Consumer<String> logStream;
	
	public LoggerOutputStream(Consumer<String> logStream) {
		this.logStream = logStream;
	}
	
	@Override
	public void write(byte @NotNull [] b) {
		logStream.accept(new String(b));
	}
	
	@Override
	public void write(byte @NotNull [] b, int off, int len) {
		logStream.accept(new String(b, off, len));
	}
	
	@Override
	public void write(int b) {
		logStream.accept(new String(new byte[]{(byte) b}));
	}
}

File: utils/src/main/kotlin/io/github/aemogie/timble/utils/logging/LoggerOutputStream.kt
package io.github.aemogie.timble.utils.logging

import java.io.OutputStream
import java.util.function.Consumer

class LoggerOutputStream(private val logStream: Consumer<String>) : OutputStream() {
	override fun write(b: ByteArray) {
		logStream.accept(String(b))
	}
	
	override fun write(b: ByteArray, off: Int, len: Int) {
		logStream.accept(String(b, off, len))
	}
	
	override fun write(b: Int) {
		logStream.accept(String(byteArrayOf(b.toByte())))
	}
}